tests/python/common/sampling/test_sampling.py

```diff
@@ -681,6 +681,117 @@ def _test3(p, replace):
         assert subg["flips"].num_edges() == 4
 
 
+def _test_sample_labors(hypersparse, prob):
+    g, hg = _gen_neighbor_sampling_test_graph(hypersparse, False)
+
+    def _test1(p):
```

Comment:
Please be specific on the naming about what case is tested here.

Reply:
I copied the test from the corresponding ones for NeighborSampling. I am not exactly sure about the exact testing scenario. The behaviors of the two samplers are pretty similar so I could do that. It is actually a bit hard to test LaborSampler because given a fanout of 2, it returns 2 neighbors per vertex in expectation. So it is hard to test for this expectation.

Reply:
I would have some concerns if the author can't provide well defined unit test cases, the algorithm is pretty complicated already :)
If we really want to make LaborSampler an "awesome version" of NeighborSampler and eventually replacing NeighborSampling in most situations, we should make sure we have 100% confident on the functional correctness and performance measurement by have concrete unit tests and regression tests.

Reply:
I do agree that the existing test is not perfect already, we should make it better.

Reply:
If there was even the slightest demand for Labor Sampler, I would have the motivation to make it %100 robust production quality. As it is, it is almost there as I believe the performance of the code is already optimized and most of the functionality will be tested for with this PR. The statistical testing for importance_sampling=0 scenario is easily doable properly but it is a bit challenging to test for importance_sampling!=0. I believe that some end to end tests where we swap NeighborSampling with LaborSampling with different hyperparameters would also be a good addition to check for accuracy of convergence etc.

Reply:
![image](https://github.com/dmlc/dgl/assets/16190101/02a07aca-7022-4a6f-b594-f88d40dffd2a)
These are my results from the paper by just replacing NeighborSampler with LaborSampler without any other change.

Reply:
You can count on me to assist with any issues related to this code. If anybody opens an issue, I will make sure that it is properly resolved or clarifications will be made.

Reply:
I will go ahead and look at the testing code to see what is going on. I will be adding comments about what is being tested for in different parts soon.